lib/spack/spack/binary_distribution.py

```diff
@@ -377,9 +394,15 @@ def make_package_relative(workdir, prefix, allow_root):
         cur_path_names.append(os.path.join(workdir, filename))
```

Comment:
The docstring should be updated to mention that this also makes absolute symlinks into relative symlinks (or rather it should say that explicitly, since the docstring is general enough to cover both cases).